ConvivaTestApp/src/androidTest/java/com/bitmovin/analytics/conviva/testapp/ConvivaAnalyticsTest.kt

```diff
@@ -0,0 +1,1570 @@
+        mockkObject(playerStateManager)
+        return playerStateManager
+    }
+
+    fun setupMocks(context: Context) {
+        clientSettingsMock = mockClientSettingsObject()
+        clientMock = mockClientObject(context, clientSettingsMock!!)
+        playerStateManagerMock = mockPlayerStateManagerObject(clientMock!!)
+
+        // Stub method return values
+        CONVIVA_SESSION_ID = Math.abs(SecureRandom().nextInt())
+        every { clientSettingsMock!!.isInitialized } returns true
+        every { clientMock!!.isInitialized } returns true
+        every { clientMock!!.createSession(any()) } returns CONVIVA_SESSION_ID
+        every { clientMock!!.playerStateManager } returns playerStateManagerMock
+    }
+
+    fun tearDownMocks () {
+        clientMock?.let { unmockkObject(it) }
+        clientSettingsMock?.let { unmockkObject(it) }
+        playerStateManagerMock?.let { unmockkObject(it)}
+    }
+
+    fun createConvivaAnalyticsObject(activity: MainActivity) : ConvivaAnalytics {
+        // Setup mocks and create ConvivaAnalytics using mock objects
+        setupMocks(activity.applicationContext)
+        return ConvivaAnalytics(
+            activity.bitmovinPlayer!!,
+            "test",
+            activity.applicationContext,
+            convivaConfig,
+            clientMock)
+    }
+
+    fun metadataOverridesDefault() : MetadataOverrides {
+        val metadata = MetadataOverrides()
+        metadata.applicationName = "Bitmovin Android Conviva test app"
+        metadata.viewerId = "awesomeViewerId"
+        metadata.assetName = "Conviva Analytics Test Asset"
+        val customInternTags: MutableMap<String, String> = HashMap()
+        customInternTags["contentType"] = "Episode"
+        metadata.custom = customInternTags
+
+        return metadata
+    }
+
+    fun metadataOverridesCustom() : MetadataOverrides {
+        val metadata = MetadataOverrides()
+        metadata.applicationName = "Bitmovin Android Conviva test app"
+        metadata.viewerId = "awesomeViewerId"
+        metadata.assetName = "Conviva Analytics Test Asset"
+        val customInternTags: MutableMap<String, String> = HashMap()
+        customInternTags["contentType"] = "Episode"
+        customInternTags["streamType"] = "CUSTOM_STREAM_TYPE"
+        customInternTags["integrationVersion"] = "CUSTOM_INTEGRATION_VERSION"
+        metadata.custom = customInternTags
+        metadata.streamType = ContentMetadata.StreamType.UNKNOWN
+        metadata.streamUrl = "CUSTOM_STREAM_URL"
+
+        return metadata
+    }
+
+    fun expectedContentMetadata(player: BitmovinPlayer,
+                                source: SourceItem,
+                                streamType: ContentMetadata.StreamType,
+                                duration: Int,
+                                overrideMetadata: MetadataOverrides,
+                                overrideCustom: Boolean = false) : ContentMetadata {
```

Comment:
I would prefer a code style like this:
```suggestion
    fun expectedContentMetadata(
        player: BitmovinPlayer,
        source: SourceItem,
        streamType: ContentMetadata.StreamType,
        duration: Int,
        overrideMetadata: MetadataOverrides,
        overrideCustom: Boolean = false
    ) : ContentMetadata {
```

Reply:
done